--- gcc-3.4.4/gcc/collect2.c.orig	2009-01-21 12:47:39.000000000 -0500
+++ gcc-3.4.4/gcc/collect2.c	2009-01-21 12:29:11.000000000 -0500
@@ -1534,7 +1534,7 @@
   if (redir)
     {
       /* Open response file.  */
-      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT);
+      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT, 0666);
 
       /* Duplicate the stdout and stderr file handles
 	 so they can be restored later.  */
